libs/mbedtls/mbedtls_stubs.c

```diff
@@ -0,0 +1,307 @@
+#include <ctype.h>
+#include <string.h>
+#include <stdio.h>
+
+#ifdef _WIN32
+#include <windows.h>
+#include <wincrypt.h>
+#endif
+
+#include <caml/mlvalues.h>
+#include <caml/alloc.h>
+#include <caml/memory.h>
+#include <caml/fail.h>
+#include <caml/callback.h>
+
+#include "mbedtls/debug.h"
+#include "mbedtls/error.h"
+#include "mbedtls/config.h"
+#include "mbedtls/ssl.h"
+#include "mbedtls/entropy.h"
+#include "mbedtls/ctr_drbg.h"
+#include "mbedtls/net_sockets.h"
+#include "mbedtls/certs.h"
+
+#define SSL(v) (mbedtls_ssl_context*)Field(v, 0)
+#define CFG(v) (mbedtls_ssl_config*)Field(v, 0)
+#define CTR(v) (mbedtls_ctr_drbg_context*)Field(v, 0)
+#define CERT(v) (mbedtls_x509_crt*)Field(v, 0)
+#define PVOID(v) (void*)Field(v, 0)
+
+void debug(void* ctx, int debug_level, const char* file_name, int line, const char* message) {
+	printf("%s:%i: %s", file_name, line, message);
+}
+
+CAMLprim value ml_mbedtls_strerror(value code) {
+	CAMLparam1(code);
+	CAMLlocal1(r);
+	char buf[128];
+	mbedtls_strerror(Int_val(code), buf, sizeof(buf));
+	r = caml_copy_string(buf);
+	CAMLreturn(r);
+}
+
+// CtrDrbg
+
+CAMLprim value ml_mbedtls_ctr_drbg_init() {
```

Comment:
This is actually fine because this is function definition instead of declaration.

https://softwareengineering.stackexchange.com/a/287002/316855
> In short, an empty parameter list in a **function declaration** indicates that the function takes an **unspecified number of parameters**, while an empty parameter list in a **function definition** indicates that the function takes **no parameters**.

Reply:
That same answer immediately follows up with:

> As far as C is concerned, you should never use an empty identifier list in a function declaration or definition. If a function is not meant to take any parameters, specify that by using void in the parameter list.